core/cmd/shell_local.go

```diff
@@ -822,6 +825,49 @@ func dropDanglingTestDBs(lggr logger.Logger, db *sqlx.DB) (err error) {
 	return
 }
 
+type failedToRandomizeTestDBSequencesError struct{}
+
+func (m *failedToRandomizeTestDBSequencesError) Error() string {
+	return "failed to randomize test db sequences"
+}
+
+// randomizeTestDBSequences randomizes sequenced table columns sequence
+// This is necessary as to avoid false positives in some test cases.
+func randomizeTestDBSequences(db *sqlx.DB) error {
+	seqRows, err := db.Query(`SELECT sequence_schema, sequence_name FROM information_schema.sequences WHERE sequence_schema = $1`, "public")
+	if err != nil {
+		return fmt.Errorf("%s: error fetching sequences: %s", failedToRandomizeTestDBSequencesError{}, err)
+	}
+
+	defer seqRows.Close()
+	for seqRows.Next() {
+		var sequenceSchema, sequenceName string
+		if err = seqRows.Scan(&sequenceSchema, &sequenceName); err != nil {
+			return fmt.Errorf("%s: failed scanning sequence rows: %s", failedToRandomizeTestDBSequencesError{}, err)
+		}
+
+		if sequenceName == "goose_migrations_id_seq" || sequenceName == "configurations_id_seq" {
+			continue
+		}
+
+		var randNum *big.Int
+		randNum, err = crand.Int(crand.Reader, utils.NewBigI(10000).ToInt())
+		if err != nil {
+			return fmt.Errorf("%s: failed to generate random number", failedToRandomizeTestDBSequencesError{})
+		}
+
+		if _, err = db.Exec(fmt.Sprintf("ALTER SEQUENCE %s.%s RESTART WITH %d", sequenceSchema, sequenceName, randNum)); err != nil {
+			return fmt.Errorf("%s: failed to alter and restart %s sequence: %w", failedToRandomizeTestDBSequencesError{}, sequenceName, err)
```

Comment:
Is it straightforward to implement regression testing for this feature? If not, nw

Reply:
Its a bit awkward to test, I tried in a couple of ways and decided that its not worth it as its hard to write a proper one.

The only worthwhile test would be to check if PrepareTestDB prepres a db with randomised sequnced tables.

Not sure if its possible to make an isolated test that calls PrepareTestDB as it is already called before all tests are ran, so calling it in a regression test would mess up other tests.


 Also its hard to measure if sequences got randomized or not, maybe a new table got added whose sequence starts from 100. For eg. I added an exception for goose, but maybe more tables like this show up and break CI for a moment.(maybe not that big of a deal)

Reply:
> Not sure if its possible to make an isolated test that calls PrepareTestDB as it is already called before all tests are ran, so calling it in a regression test would mess up other tests.

Oh, huh - does this mean before you implemented random sequences, `PrepareTestDB` was not idempotent?

Reply:
Randomising sequences didn't change much it just makes it unlikely to get a false positive in some tests [eg.](https://github.com/smartcontractkit/chainlink/pull/10781). From my understanding we have db isolation on transaction level between most tests, unless heavyweight pckg is explicitly used. 
This actually reminded me that I need to add rand sequences to that heavyweight package as well